### PR TITLE
Update the `xxhash` and `hashlib` stubs

### PR DESCRIPTION
**Pull request by BvB93**
This PR updates a few annotations in the third-party `xxhash` stubs and, updates that, due to inheritance, somewhat extend to the builtin `hashlib` library.

The changes introduced herein are as following:
* `hashlib._Hash` attributes are now get-only properties.
* `hashlib._Hash.copy` now respects subclasses.
* `xxhash._IntDigestHash` subclasses are marked as `final`.
* The various function in `xxhash` can now take `SupportsIndex`-implementing objects, rather than just `int`.

### CODE REVIEW

File: stdlib/hashlib.pyi [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 
 class _Hash(object):
-    digest_size: int
-    block_size: int
-
+    @property
+    def digest_size(self) -> int: ...
+    @property
+    def block_size(self) -> int: ...
     # [Python documentation note] Changed in version 3.4: The name attribute has
```
Comment (JelleZijlstra) [inline stdlib/hashlib.pyi:10]: This comment can be removed, we don't care about Python < 3.4 anyway.

File: stubs/xxhash/xxhash.pyi [hunk 2]
```diff
@@ -10,30 +11,39 @@ VERSION: str
...
-def xxh64_digest(input: ReadableBuffer = ..., seed: int = ...) -> bytes: ...
-def xxh64_intdigest(input: ReadableBuffer = ..., seed: int = ...) -> int: ...
-def xxh64_hexdigest(input: ReadableBuffer = ..., seed: int = ...) -> str: ...
-def xxh3_64_digest(input: ReadableBuffer = ..., seed: int = ...) -> bytes: ...
-def xxh3_64_intdigest(input: ReadableBuffer = ..., seed: int = ...) -> int: ...
-def xxh3_64_hexdigest(input: ReadableBuffer = ..., seed: int = ...) -> str: ...
-def xxh3_128_digest(input: ReadableBuffer = ..., seed: int = ...) -> bytes: ...
-def xxh3_128_intdigest(input: ReadableBuffer = ..., seed: int = ...) -> int: ...
-def xxh3_128_hexdigest(input: ReadableBuffer = ..., seed: int = ...) -> str: ...
+def xxh32_digest(input: ReadableBuffer = ..., seed: SupportsIndex = ...) -> bytes: ...
```
Comment (JelleZijlstra) [inline stubs/xxhash/xxhash.pyi:35]: The input argument seems to be required.
```suggestion
def xxh32_digest(input: ReadableBuffer, seed: SupportsIndex = ...) -> bytes: ...
```
I'm guessing that's also true for the other digest functions but I didn't check.

**Review by JelleZijlstra — COMMENTED** [review]
Thanks! I noticed a couple of other issues that were already present in the stubs; would you mind fixing them too?

File: stdlib/hashlib.pyi (hunk 1 above)
Comment (BvB93) [reply stdlib/hashlib.pyi]: Done as of https://github.com/python/typeshed/pull/6031/commits/e4291458236d779e95291b0e1643fec53790fa19.

File: stubs/xxhash/xxhash.pyi (hunk 2 above)
Comment (BvB93) [reply stubs/xxhash/xxhash.pyi]: Done as of https://github.com/python/typeshed/pull/6031/commits/5a2323743096afa71870a828fa5f2f07acf236ba; it seems to be true for all functions indeed.

File: stubs/xxhash/xxhash.pyi [hunk 3]
```diff
@@ -10,30 +11,39 @@ VERSION: str
...
+def xxh32_hexdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> str: ...
+def xxh64_digest(input: ReadableBuffer, seed: SupportsIndex = ...) -> bytes: ...
+def xxh64_intdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> int: ...
+def xxh64_hexdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> str: ...
+def xxh3_64_digest(input: ReadableBuffer, seed: SupportsIndex = ...) -> bytes: ...
+def xxh3_64_intdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> int: ...
+def xxh3_64_hexdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> str: ...
+def xxh3_128_digest(input: ReadableBuffer, seed: SupportsIndex = ...) -> bytes: ...
+def xxh3_128_intdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> int: ...
+def xxh3_128_hexdigest(input: ReadableBuffer, seed: SupportsIndex = ...) -> str: ...
```
Comment (Akuli) [inline stubs/xxhash/xxhash.pyi:46]: Apparently the input can be a string. We can fix that in this PR or in a separate PR.

Comment (BvB93) [reply stubs/xxhash/xxhash.pyi]: No need, done as of https://github.com/python/typeshed/pull/6031/commits/fe55b5639f820fa3095c5564cdac0aed9ec2b745.

**Review by Akuli — APPROVED** [review]

**Pull request merged**